### PR TITLE
CA-185733: RPU XenCenter stuck on upgrade screen if connected as XS Pool Admin

### PR DESCRIPTION
**Pull request by phusl**
Before xapi_session.do_external_auth, we should check the external auth service status at first.

Signed-off-by: Phus Lu <phus.lu@citrix.com>

### CODE REVIEW

File: ocaml/xapi/xapi_session.ml [hunk 1]
```diff
@@ -415,8 +415,23 @@ let login_with_password ~__context ~uname ~pwd ~version ~originator = wipe_param
 					debug "Failed to locally authenticate user %s from %s: %s" uname (Context.get_origin __context) msg;
 					
 					(* 2. then against the external auth service *)
-					
-					(* 2.1. we first authenticate the user using the external authentication plugin *)
+					(* 2.1. we first check the external auth service status *)
+					let rec waiting_event_hook_auth_on_xapi_initialize_succeeded seconds =
+						if !Xapi_globs.event_hook_auth_on_xapi_initialize_succeeded then
+							0
```
Comment (johnelse) [inline ocaml/xapi/xapi_session.ml]: Why not return `()` here? That way you won't need the `ignore` on line 443.

**Comment by johnelse** [comment]
Other than the minor comment above, this logic looks fine to me - thanks!

File: ocaml/xapi/xapi_session.ml (hunk 1 above)
Comment (phusl) [inline ocaml/xapi/xapi_session.ml]: Thank you for the comments, I updated this PR and verified in trunk-ca-185733/108093.

**Comment by phusl** [comment]
@johnelse Thank you for your comments, I updated the PR.

**Comment by johnelse** [comment]
Great, thanks! I'll merge this now.

**Pull request merged**